daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/XMLUtils.scala

```diff
@@ -1488,8 +1499,24 @@ Differences were (path, expected, actual):
         val contextURI = optContextURI.get
         val optResolvedRelative = None
           .orElse {
+            val contextURIDFP = contextURI.diagnosticFilepath
+            val diagnosticFilepathPaths = if (contextURIDFP.startsWith("/")) {
+              new File(contextURIDFP).toPath
+            } else {
+              Paths.get(Misc.stripJarPrefix(contextURIDFP))
+            }
+            val relativeDiagnosticFilepath =
+              if (new File(uri.getPath).compareTo(diagnosticFilepathPaths.toFile) == 0) { // paths are the same
+                diagnosticFilepathPaths.toString
+              } else {
+                diagnosticFilepathPaths.resolveSibling(uri.getPath).toString
+              }
```

Comment:
We need some comments about what is going on here. I'm not sure I understand why we need this.

Seem this logic is because sometimes the diagnosticFilePath isn't a Path, like it's a URI or something? Maybe if we require that it must be a Path, mentioned in another comment, then this special case  goes away? If diagnosticFilePath is just a Path, we can always just call resolveSibling here and it should just work?